### DISCUSSION TITLE
Please help

### DISCUSSION THREAD

**Discussion by pulkitchhabra19** [jax-ml/jax#13564] [Q&A]
Here is a part of my code.
```python
class xx
  ...

  def abc(self, features)
    ...
    t0 = features["t0"]
    t = []
    for t_end in features["t"]:
      ts = jnp.arange(jnp.ceil(t0), jnp.ceil(t_end) + 1)
      ...
      ys = odeint(func, y0, ts, features)
      t.extend(ts)
      t0 = t[-1]
```
This works fine for the first complete run of the for loop. But next time I try to run it, I get an error because then t_end is a abstract tensor and it complains about using an abstract tensor where a static value is needed. Can you advise how I can work around that?

I read about static_argnames but could not figure out how I can make features static when working with odeint.
I also tried pure_callback with t_end to get it in the form of a numpy array. but nothing seems to work. Can't I perform operations on features if I pass it to odeint as input argument?? Any help would be appreciated.

**Comment by jakevdp** [comment]
Thanks for the question! The issue is that JAX cannot create dynamically-shaped arrays within transformations like `jit`, and `jnp.arange(jnp.ceil(t0), jnp.ceil(t_end) + 1)` will have a dynamic size (i.e. a size dependent on the contents of `t0` and `t_end`, which given your calls to `jnp.ceil` are presumably arrays).

If you want to use JAX, you'll have to find a different way to express your computation that does not rely on creating dynamically-sized arrays. If you can edit your code snippet to add more complete information about what you're passing to these functions, we might be able to give more pointed advice.
